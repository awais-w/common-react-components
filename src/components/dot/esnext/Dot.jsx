import React from 'react';
import PropTypes from 'prop-types';
import bem from '@argos/utils/esnext/bem';
import './Dot.scss';

const cn = bem({ prefix: 'cmc-', block: 'dot' });

const Dot = ({ color, value }) => (
  <div className={cn(null, color)}>
    <span>{value}</span>
  </div>
);

Dot.propTypes = {
  color: PropTypes.oneOf([undefined, 'white', 'red', 'yellow', 'blue', 'black', 'grey']),
  value: PropTypes.string,
};

Dot.defaultProps = {
  color: 'black',
};

export default Dot;
