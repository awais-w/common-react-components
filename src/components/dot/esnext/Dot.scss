@import '@cmc/common-styles/src/scss/_variables.scss';

.cmc-dot {
  display: flex;
  justify-content: center;
  align-items: center;
  height: $large;
  width: $large;
  border-radius: 50%;
  color: $color-white;
  @include font-standard;
  &--red {
    background-color: $color-red;
  }
  &--yellow {
    background-color: $color-star-stroke;
  }
  &--blue {
    background-color: $primary-color;
  }
  &--black {
    background-color: $color-black;
  }
  &--grey {
    background-color: $color-grey-dark;
  }
}
