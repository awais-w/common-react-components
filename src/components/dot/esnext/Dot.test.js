import React from 'react';
import { shallow } from 'enzyme';
import Dot from './Dot';

const setup = ({ props }) => shallow(<Dot {...props} />);

describe('Dot', () => {
  it('should match default snapshot', () => {
    const component = setup({ props: {} });
    expect(component).toMatchSnapshot();
  });

  it('renders correct color', () => {
    const component = setup({ props: { color: 'yellow' } });
    expect(component.find('.cmc-dot--yellow')).toExist();
  });

  it('renders correct text', () => {
    const component = setup({ props: { value: '5' } });
    expect(component.find('span').text()).toBe('5');
  });
});
