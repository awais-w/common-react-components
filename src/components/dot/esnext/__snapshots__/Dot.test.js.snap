// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dot should match default snapshot 1`] = `
<div
  className="cmc-dot cmc-dot--black"
>
  <span />
</div>
`;
